### PR TITLE
reef: cephadm/nvmeof: scrape nvmeof prometheus endpoint

### PR DESCRIPTION
**Pull request by avanthakkar**
backport tracker: https://tracker.ceph.com/issues/64635

---

backport of https://github.com/ceph/ceph/pull/55710
parent tracker: https://tracker.ceph.com/issues/64536

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/main/src/script/ceph-backport.sh

### CODE REVIEW

File: src/pybind/mgr/cephadm/templates/services/nvmeof/ceph-nvmeof.conf.j2 [hunk 1]
```diff
@@ -7,6 +7,12 @@ port = {{ port }}
 enable_auth = {{ spec.enable_auth }}
 state_update_notify = True
 state_update_interval_sec = 5
+min_controller_id = {{ spec.min_controller_id }}
+max_controller_id = {{ spec.max_controller_id }}
+enable_spdk_discovery_controller = {{ spec.enable_spdk_discovery_controller }}
+enable_prometheus_exporter = True
+prometheus_exporter_ssl = False
+prometheus_port = 10008
```
Comment (adk3798) [inline src/pybind/mgr/cephadm/templates/services/nvmeof/ceph-nvmeof.conf.j2:15]: make check failure is actually related
```
FAILED cephadm/tests/test_services.py::TestNVMEOFService::test_nvmeof_config
```

```
E           Expected: _run_cephadm('test', 'nvmeof.testpool.test.qwert', ['_orch', 'deploy'], [], stdin='{"fsid": "fsid", "name": "nvmeof.testpool.test.qwert", "image": "", "deploy_arguments": [], "params": {"tcp_ports": [5500, 4420, 8009]}, "meta": {"service_name": "nvmeof.testpool", "ports": [5500, 4420, 8009], "ip": null, "deployed_by": [], "rank": null, "rank_generation": null, "extra_container_args": null, "extra_entrypoint_args": null}, "config_blobs": {"config": "", "keyring": "[client.nvmeof.testpool.test.qwert]\\nkey = None\\n", "files": {"ceph-nvmeof.conf": "# This file is generated by cephadm.\\n[gateway]\\nname = client.nvmeof.testpool.test.qwert\\ngroup = mygroup\\naddr = 192.168.100.100\\nport = 5500\\nenable_auth = False\\nstate_update_notify = True\\nstate_update_interval_sec = 5\\nmin_controller_id = 1\\nmax_controller_id = 65519\\nenable_spdk_discovery_controller = False\\nenable_prometheus_exporter = True\\nprometheus_exporter_ssl = False\\nprometheus_port = 10008\\n\\n[ceph]\\npool = testpool\\nconfig_file = /etc/ceph/ceph.conf\\nid = nvmeof.testpool.test.qwert\\n\\n[mtls]\\nserver_key = ./server.key\\nclient_key = ./client.key\\nserver_cert = ./server.crt\\nclient_cert = ./client.crt\\n\\n[spdk]\\ntgt_path = /usr/local/bin/nvmf_tgt\\nrpc_socket = /var/tmp/spdk.sock\\ntimeout = 60\\nlog_level = WARN\\nconn_retries = 10\\ntransports = tcp\\ntransport_tcp_options = {\\"in_capsule_data_size\\": 8192, \\"max_io_qpairs_per_ctrlr\\": 7}\\ntgt_cmd_extra_args = --cpumask=0xFF --msg-mempool-size=524288\\n"}}}')
E           Actual: _run_cephadm('test', 'osd', 'ceph-volume', ['--', 'inventory', '--format=json-pretty', '--filter-for-batch'], no_fsid=False, error_ok=False, image='', log_output=False)
```

**Comment by adk3798** [comment]
@avanthakkar given the conflicts around backporting this, I've opened my own version of this backport at https://github.com/ceph/ceph/pull/56108 where I think the tests should be passing. If that's okay with you we can close this and use that backport (and if you could review it, it would be appreciated)

**Comment by avanthakkar** [comment]
> @avanthakkar given the conflicts around backporting this, I've opened my own version of this backport at #56108 where I think the tests should be passing. If that's okay with you we can close this and use that backport (and if you could review it, it would be appreciated)

Sure, sounds good to me!

**Comment by adk3798** [comment]
closing in favor of (the already merged) https://github.com/ceph/ceph/pull/56108

**Pull request closed without merging**